src/main/scala/devices/debug/Debug.scala

```diff
@@ -688,6 +687,11 @@ class TLDebugModuleInner(device: Device, getNComponents: () => Int, beatBytes: I
     val nHaltGroups = if ((nComponents > 1) | (nExtTriggers > 0)) cfg.nHaltGroups
       else 0  // no halt groups possible if single hart with no external triggers
 
+    val hartSelFuncs = if (getNComponents() > 1) p(DebugModuleHartSelKey) else DebugModuleHartSelFuncs(
+      hartIdToHartSel = (x) => UInt(0),
+      hartSelToHartId = (x) => x
```

Comment:
wouldnt this also need to be 0?

Reply:
No, that would have no effect.  This code only applies to a single-core system, so hartSel is always 0.  0 => 0 in either case.

Reply:
ok makes sense